discretize/TreeMesh.py

```diff
@@ -1057,195 +1130,242 @@ def _hanging(self, force=False):
-                self._hangingFx[self._fx2i[self._index([p[0]    , p[1] + w, sl])]] = ([self._fx2i[fx], chy1 / A], )
+                self._hangingFx[self._fx2i[test]] = (
+                    [self._fx2i[fx], chy0 / A],
+                )
+                self._hangingFx[
+                    self._fx2i[self._index([p[0], p[1] + w, sl])]
+                ] = (
+                    [self._fx2i[fx], chy1 / A],
+                )
 
                 n0, n1 = fx, self._index([p[0], p[1] + 2*w, p[-1]])
-                self._hangingN[self._n2i[test                                   ]] = ([self._n2i[n0], 1.0], )
-                self._hangingN[self._n2i[self._index([p[0]    , p[1] +   w, sl])]] = ([self._n2i[n0], 1.0 - chy0 / A], [self._n2i[n1], 1.0 - chy1 / A])
-                self._hangingN[self._n2i[self._index([p[0]    , p[1] + 2*w, sl])]] = ([self._n2i[n1], 1.0], )
+                self._hangingN[self._n2i[test]] = ([self._n2i[n0], 1.0], )
+                self._hangingN[
+                    self._n2i[self._index([p[0], p[1] + w, sl])]
+                ] = (
+                    [self._n2i[n0], 1.0 - chy0 / A],
+                    [self._n2i[n1], 1.0 - chy1 / A]
+                )
+                self._hangingN[
+                    self._n2i[self._index([p[0], p[1] + 2*w, sl])]
+                ] = ([self._n2i[n1], 1.0], )
 
             elif self.dim == 3:
 
-                chy0 = self._cellH([p[0]    , p[1]    , p[2]    , sl])[1]
-                chy1 = self._cellH([p[0]    , p[1] + w, p[2]    , sl])[1]
-                chz0 = self._cellH([p[0]    , p[1]    , p[2]    , sl])[2]
-                chz1 = self._cellH([p[0]    , p[1]    , p[2] + w, sl])[2]
+                chy0 = self._cellH([p[0], p[1], p[2], sl])[1]
+                chy1 = self._cellH([p[0], p[1] + w, p[2], sl])[1]
+                chz0 = self._cellH([p[0], p[1], p[2], sl])[2]
+                chz1 = self._cellH([p[0], p[1], p[2] + w, sl])[2]
                 lenY = chy0 + chy1
                 lenZ = chz0 + chz1
                 A = lenY * lenZ
 
                 ey0 = fx
-                ey1 = self._index([p[0], p[1]      , p[2] + 2*w, p[-1]])
+                ey1 = self._index([p[0], p[1], p[2] + 2*w, p[-1]])
                 ez0 = fx
-                ez1 = self._index([p[0], p[1] + 2*w, p[2]      , p[-1]])
+                ez1 = self._index([p[0], p[1] + 2*w, p[2], p[-1]])
 
-                n0  = fx
-                n1  = self._index([p[0], p[1] + 2*w, p[2]      , p[-1]])
-                n2  = self._index([p[0], p[1]      , p[2] + 2*w, p[-1]])
-                n3  = self._index([p[0], p[1] + 2*w, p[2] + 2*w, p[-1]])
+                n0 = fx
+                n1 = self._index([p[0], p[1] + 2*w, p[2], p[-1]])
+                n2 = self._index([p[0], p[1], p[2] + 2*w, p[-1]])
+                n3 = self._index([p[0], p[1] + 2*w, p[2] + 2*w, p[-1]])
 
                 i000 = test
-                i010 = self._index([p[0], p[1] +   w, p[2]      , sl])
```

Comment:
I guess I will comment that PEP8 in this case makes it a lot harder to see what is going on in some of the more detailed code.

For example:
```python
i010 = self._index([p[0], p[1] +   w, p[2]      , sl])
i001 = self._index([p[0], p[1]      , p[2] +   w, sl])
i011 = self._index([p[0], p[1] +   w, p[2] +   w, sl])
i020 = self._index([p[0], p[1] + 2*w, p[2]      , sl])
i021 = self._index([p[0], p[1] + 2*w, p[2] +   w, sl])
i002 = self._index([p[0], p[1]      , p[2] + 2*w, sl])
i012 = self._index([p[0], p[1] +   w, p[2] + 2*w, sl])
i022 = self._index([p[0], p[1] + 2*w, p[2] + 2*w, sl])
```

versus:

```python
i010 = self._index([p[0], p[1] + w, p[2], sl])
i001 = self._index([p[0], p[1], p[2] + w, sl])
i011 = self._index([p[0], p[1] + w, p[2] + w, sl])
i020 = self._index([p[0], p[1] + 2*w, p[2], sl])
i021 = self._index([p[0], p[1] + 2*w, p[2] + w, sl])
i002 = self._index([p[0], p[1], p[2] + 2*w, sl])
i012 = self._index([p[0], p[1] + w, p[2] + 2*w, sl])
i022 = self._index([p[0], p[1] + 2*w, p[2] + 2*w, sl])
```

I guess ultimately this should have been done with an index?

```python
offsets = [0, 0], [w , 0], [0, w], [w, w]
i1s = [
    self._index(p[0], p[1] + offset[0], p[2] + offset[1])
    for offset in offsets
]
```
But then you have to remember the index?